src/components/AppConfig/Vector.tsx

```diff
@@ -0,0 +1,186 @@
+import React from "react";
+
+import { Field, FieldSet, Input, Select, Switch, useStyles2 } from "@grafana/ui";
+
+import { testIds } from "components/testIds";
+import { getStyles } from "./AppConfig";
+
+export interface VectorSettings {
+  // Whether the vector service should be enabled.
+  enabled?: boolean;
+  // The model used by the embedder and for embeddings stored in the store.
+  model?: string;
+  // Embedder settings.
+  embed?: EmbedderSettings;
+  // Store settings.
+  store?: StoreSettings;
+};
+
+interface OpenAIEmbedderSettings {
+  url?: string;
+};
+
+interface EmbedderSettings {
+  type?: string;
+  openai?: OpenAIEmbedderSettings;
+};
+
+interface QdrantSettings {
+  address?: string;
+  secure?: boolean;
+}
+
+interface GrafanaVectorAPISettings {
+  url?: string;
+}
+
+interface StoreSettings {
+  type?: string;
+  qdrant?: QdrantSettings;
+  grafanaVectorAPI?: GrafanaVectorAPISettings;
+}
+
+interface Props<T> {
+  settings?: T;
+  onChange: (settings: T) => void;
+}
+
+export function VectorConfig({ settings, onChange }: Props<VectorSettings>) {
+  return (
+    <FieldSet label="Vector Settings">
+      <Field label="Model" description="The model used by the embedder and for embeddings stored in the store">
+        <Input
+          width={60}
+          name="model"
```

Comment:
Good as-is
Maybe for the future, we could add some validation such as
- `text-embedding-ada-002` only available if using OpenAI / Azure OpenAI
- limit the models to only ones we've tried and tested (if using custom embedder) 🤔

Reply:
Yeah some kind of validation would be great, I think doing it via the backend health check would make most sense though since the set of models is pretty much unbounded and we don't want to be keeping our select options constantly updated.

I'm thinking that the health check would do a test chat completions run, test embedding, and test vector search, and we'd validate based on that.